test/code-cache/test-code-cache-generator.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+// This test verifies that the binary is compiled with code cache and the
+// cache is used when built in modules are compiled.
+
+const common = require('../common');
+if (!common.enoughTestCpu || !common.enoughTestMem) {
+  common.skip('This test is resource-intensive');
+}
+const tmpdir = require('../common/tmpdir');
+const { spawnSync } = require('child_process');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const generator = path.join(
+  __dirname, '..', '..', 'tools', 'generate_code_cache.js'
+);
+tmpdir.refresh();
+const dest = path.join(tmpdir.path, 'cache.cc');
+
+const child = spawnSync(
+  process.execPath,
+  ['--expose-internals', generator, dest]
+);
+assert.ifError(child.error);
+
+const stat = fs.statSync(dest);
+assert(stat.isFile());
```

Comment:
Prefer `assert.ok`
Is there anything else we can test about the file? Maybe that it has the string `static uint8_t internal_bootstrap_cache_raw[]`?